### PR TITLE
BUILD: Refactor spirv submodule

### PR DESCRIPTION
**Pull request by p01arst0rm**
combined spirv* directories into a single directory structure,
and moved them to `3rdparty/`

### CODE REVIEW

**Comment by p01arst0rm** [comment]
cc @hcorion

File: 3rdparty/SPIRV/spirv.vcxproj [hunk 1]
```diff
@@ -1,69 +1,69 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (Megamouse) [inline 3rdparty/SPIRV/spirv.vcxproj:1]: what's with all these random line changes?

File: 3rdparty/SPIRV/spirv.vcxproj [hunk 2]
```diff
@@ -1,69 +1,69 @@
...
+    <ProjectConfiguration Include="Release|x64">
+      <Configuration>Release</Configuration>
+      <Platform>x64</Platform>
+    </ProjectConfiguration>
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <ProjectGuid>{4CBD3DDD-5555-49A4-A44D-DD3D8CB516A1}</ProjectGuid>
+    <Keyword>MakeFileProj</Keyword>
+  </PropertyGroup>
+  <Import Project="$(SolutionDir)\buildfiles\msvc\common_default.props" />
```
Comment (Megamouse) [inline 3rdparty/SPIRV/spirv.vcxproj:17]: SolutionDir does not work in azure builds in this file.
so you can't commit this.

File: rpcs3/rpcs3.vcxproj [hunk 3]
```diff
@@ -140,7 +140,7 @@
...
-      <AdditionalLibraryDirectories>..\3rdparty\OpenAL\libs\Win64;..\3rdparty\glslang\hlsl\Debug;..\3rdparty\glslang\SPIRV\Debug;..\3rdparty\glslang\OGLCompilersDLL\Debug;..\3rdparty\glslang\glslang\OSDependent\Windows\Debug;..\3rdparty\glslang\glslang\Debug;..\Vulkan\spirv-tools-build\source\opt\Debug;..\3rdparty\XAudio2Redist\libs;..\3rdparty\discord-rpc\lib;..\lib\$(CONFIGURATION)-$(PLATFORM);$(QTDIR)\lib;%(AdditionalLibraryDirectories);$(VULKAN_SDK)\Lib</AdditionalLibraryDirectories>
+      <AdditionalLibraryDirectories>..\3rdparty\OpenAL\libs\Win64;..\3rdparty\glslang\hlsl\Debug;..\3rdparty\glslang\SPIRV\Debug;..\3rdparty\glslang\OGLCompilersDLL\Debug;..\3rdparty\glslang\glslang\OSDependent\Windows\Debug;..\3rdparty\glslang\glslang\Debug;..\3rdparty\spirv-tools-build\source\opt\Debug;..\3rdparty\XAudio2Redist\libs;..\3rdparty\discord-rpc\lib;..\lib\$(CONFIGURATION)-$(PLATFORM);$(QTDIR)\lib;%(AdditionalLibraryDirectories);$(VULKAN_SDK)\Lib</AdditionalLibraryDirectories>
```
Comment (Megamouse) [inline rpcs3/rpcs3.vcxproj:143]: is this correct?

File: 3rdparty/CMakeLists.txt [hunk 4]
```diff
@@ -82,8 +82,9 @@ endif()
 add_subdirectory(hidapi)
 
 
-# glslang
+# Vulkan
 add_subdirectory(glslang)
+add_subdirectory(SPIRV)
```
Comment (hcorion) [inline 3rdparty/CMakeLists.txt:87]: Prolly want EXCLUDE_FROM_ALL here if we did it for the Vulkan folder

File: 3rdparty/SPIRV/spirv.vcxproj (hunk 2 above)
Comment (p01arst0rm) [reply 3rdparty/SPIRV/spirv.vcxproj:17]: alright, will revert

File: rpcs3/rpcs3.vcxproj (hunk 3 above)
Comment (p01arst0rm) [reply rpcs3/rpcs3.vcxproj:143]: compiles correctly on all platforms for me; what specifically looks wrong?

File: .gitmodules [hunk 5]
```diff
@@ -16,11 +16,11 @@
 	url = https://github.com/KhronosGroup/glslang.git
 	ignore = dirty
 [submodule "Vulkan/spirv-tools"]
-	path = Vulkan/spirv-tools
+	path = 3rdparty/SPIRV/spirv-tools
```
Comment (hcorion) [inline .gitmodules:19]: It bugs me that the SPIRV folder is all caps but spirv-tools is all lowercase.
It should either be SPIRV/SPIRV-Tools or spirv/spirv-tools. I'm leaning towards the former for consistency with upstream.

File: rpcs3/rpcs3.vcxproj (hunk 3 above)
Comment (Megamouse) [reply rpcs3/rpcs3.vcxproj:143]: I thought you had renamed the spirv-tools-build directory

Comment (p01arst0rm) [reply rpcs3/rpcs3.vcxproj:143]: ahh nvm i see it; the debug wasnt changed to 3rdparty\SPIRV\source\opt\Debug. I
never tried a debug build as it's currently broken anyway. Will fix, thanks :)

File: 3rdparty/SPIRV/spirv.vcxproj (hunk 1 above)
Comment (p01arst0rm) [reply 3rdparty/SPIRV/spirv.vcxproj:1]: probably an automated change by vs i didnt notice :) will revert.

File: .gitignore [hunk 6]
```diff
@@ -40,7 +40,7 @@
 /3rdparty/llvm_build
 /Vulkan/Vulkan-build
```
Comment (hcorion) [inline .gitignore:41]: Can get rid of this line as well I think

**Review by hcorion — CHANGES_REQUESTED** [review]

File: 3rdparty/SPIRV/spirv.vcxproj [hunk 7]
```diff
@@ -16,7 +16,7 @@
   </PropertyGroup>
   <Import Project="..\..\buildfiles\msvc\common_default.props" />
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
-  <Import Project="..\..\buildfiles\msvc\common_default_macros.props" />
+  <Import Project="$(SolutionDir)\buildfiles\msvc\common_default_macros.props" />
```
Comment (Megamouse) [inline 3rdparty/SPIRV/spirv.vcxproj:19]: Solution die still doesn't work in this file

Comment (Megamouse) [reply 3rdparty/SPIRV/spirv.vcxproj:19]: Weird. I had some old GitHub cache. Nvm

File: 3rdparty/SPIRV/spirv.vcxproj [hunk 8]
```diff
@@ -1,69 +1,69 @@
...
+  </ImportGroup>
+  <ImportGroup Label="PropertySheets" Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
+    <Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
+  </ImportGroup>
+  <PropertyGroup Label="UserMacros">
+    <CmakeGenerator>"Visual Studio $(VisualStudioVersion.Substring(0,2))"</CmakeGenerator>
+    <CmakeReleaseCLI>cmake -G $(CmakeGenerator) -A x64 -DCMAKE_BUILD_TYPE="Release" -DSPIRV_WERROR=OFF -DSPIRV-Headers_SOURCE_DIR=$([System.IO.Path]::GetFullPath('$(MSBuildThisFileDirectory)/spirv-headers')) spirv-tools</CmakeReleaseCLI>
+    <CmakeDebugCLI>cmake -G $(CmakeGenerator) -A x64 -DCMAKE_BUILD_TYPE="Debug" -DSPIRV_WERROR=OFF -DSPIRV-Headers_SOURCE_DIR=$([System.IO.Path]::GetFullPath('$(MSBuildThisFileDirectory)/spirv-headers')) spirv-tools</CmakeDebugCLI>
+    <PropsAbsPath>$([System.IO.Path]::GetFullPath('$(SolutionDir)\buildfiles\msvc\common_default.props'))</PropsAbsPath>
```
Comment (Megamouse) [inline 3rdparty/SPIRV/spirv.vcxproj:43]: Solution die doesn't work in this file

**Comment by Megamouse** [comment]
you still need to change these 3 files here:
image

**Comment by p01arst0rm** [comment]
> you still need to change these 3 files here:
> image

yep, was just fixing the bad line endings in the file. not finished changing yet :)

**Comment by Megamouse** [comment]
don't bother if visual studio did it automatically

**Comment by p01arst0rm** [comment]
okay. should all be fixed now.

File: .cirrus.yml [hunk 9]
```diff
@@ -40,8 +40,8 @@ windows_task:
 #   - choco install -y python # Needed for SPIRV, use either this or make a new Docker image
 #  spirv_script:
 #   - export PATH=${PATH}:"C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\MSBuild\Current\Bin"
-#   - cd "${CIRRUS_WORKING_DIR}/Vulkan/spirv-tools-build"
-#   - msbuild.exe spirv-tools-build.vcxproj //p:Configuration=Release //m
+#   - cd "${CIRRUS_WORKING_DIR}/3rdparts/SPIRV"
```
Comment (Megamouse) [inline .cirrus.yml:43]:
```suggestion
#   - cd "${CIRRUS_WORKING_DIR}/3rdparty/SPIRV"
```

File: BUILDING.md [hunk 10]
```diff
@@ -115,7 +115,7 @@ You may want to download the precompiled [LLVM libs](https://github.com/RPCS3/ll
 If you're not using the precompiled libs, build the following projects in *__BUILD_BEFORE* folder by right-clicking on a project > *Build*.:
 * glslang
 * **Either** llvm_build **or** llvm_build_clang_cl
-* spirv-tools-build
+* SPIRV
```
Comment (Megamouse) [inline BUILDING.md:118]:
```suggestion
* spirv
```

**Review by hcorion — APPROVED** [review]
Looks good

**Pull request merged**